src/iron.rs

```diff
@@ -58,7 +75,13 @@ impl<H: Handler> Iron<H> {
         let SocketAddr { ip, port } = sock_addr.clone();
         self.addr = Some(sock_addr);
 
-        Ok(try!(Server::http(ip, port).listen_threads(self, threads)))
+        let server = match self.security {
+            Security::Insecure => Server::http(ip, port),
+            Security::Secure { ref certificate, ref key } =>
+                Server::https(ip, port, certificate.clone(), key.clone())
```

Comment:
No, one enum would be used, called `Protocol`, completely equivalent to todays `Security`. The difference I am suggesting is that we pass the Protocol in through `listen` or `listen_with` rather than in `new` or some other constructor.